10-Reflection/Reflection.Tasks/CommonTasks.cs

```diff
@@ -61,7 +92,35 @@ public static class CommonTasks
         /// <param name="value">assigned value</param>
         public static void SetPropertyValue(this object obj, string propertyPath, object value) {
             // TODO : Implement SetPropertyValue method
-            throw new NotImplementedException();
+            //throw new NotImplementedException();
+
+            Type type;
+            PropertyInfo field;
+            Object buff = obj;
+
+            var propertys = propertyPath.Split('.');
+            foreach (var p in propertys.Take(propertys.Count() - 1))
+            {
+                
+                type = obj.GetType();
+                field = type.GetProperty(p, BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+                obj = field.GetValue(obj);
+            }
+
+            type = obj.GetType();
+            field = type.GetProperty(propertys.Last(), BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+
+            while (true)
```

Comment:
while true is generally a bad practice. Try come up wtih the proper exit condition